aten/src/ATen/core/boxing/KernelFunction.cpp

```diff
@@ -23,6 +23,8 @@ void ambiguous_autogradother_kernel(OperatorKernel*, const OperatorHandle& op, S
   TORCH_INTERNAL_ASSERT(0,
     op.operator_name(), " has kernels registered to both Math and a backend mapped to AutogradOther. "
     "This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). "
+    "If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` "
+    "before model.forward().\n"
```

Comment:
I'd actually move this new info to the end of the text, so the "If it's intended..." part relates more clearly to the AutogradOther kernel mentioned above.

Reply:
Also, is it a problem that the advice is C++ specific?